scipy/signal/fir_filter_design.py

```diff
@@ -1260,3 +1262,103 @@ def minimum_phase(h, method='homomorphic', n_fft=None):
         h_minimum = h_temp.real
     n_out = n_half + len(h) % 2
     return h_minimum[:n_out]
+
+
+def firgammatone(freq, order=4, fs=16000, numtaps=240, scale=True):
+    """
+    FIR gammatone filter design.
+
+    This function computes the coefficients of a Nth order
+    finite impuse response Gammatone filter.
+
+    Parameters
+    ----------
+    freq : float
+        Center frequency of the filter in Hz.
+    order : int, optional
+        The order of the filter. Default is 4 to model the human
+        auditory system.
+    fs : int, optional
+        The sampling frequency of the signal. `freq` must be between
+        0 and ``fs / 2``. Default is 16000.
+    numtaps : int, optional
+        Length of the filter. Optimal ratio between numtaps
+        and fs is 0.015. Default is 240.
+    scale : bool, optional
+        Set to True to scale the filter so that the frequency
+        response is exactly unity at a certain frequency.
+
+    Returns
+    -------
+    f : (numtaps,) ndarray
+        Coefficients of length `numtaps` FIR gammatone filter.
+
+    Raises
+    ------
+    ValueError
+        If `freq` is not an integer or float,
+        if `freq` is <= 0 or >= ``fs / 2``,
+        if `order` is not an integer,
+        if `order` is <= 0 or > 24,
+        if `fs` is not an integer,
+        if `numtaps is not an integer,
+        if `scale` is not a boolean.
+
+
+    See Also
+    --------
+    iirgammatone
+    firwin
+    firwin2
+
+    References
+    ----------
+    Slaney, Malcolm, "An Efficient Implementation of the
+    Patterson–Holdsworth Auditory Filter Bank", pp.3-8.
+
+    Examples
+    --------
+    240-sample Gammatone filter centered at 440 Hz
+
+    >>> from scipy import signal
+    >>> freq = 440
+    >>> signal.firgammatone(freq, numtaps=16)
+    array([ 0.00000000e+00,  2.68586728e-07,  1.97702315e-06,  5.90239274e-06,
+        1.18022128e-05,  1.82536512e-05,  2.26825900e-05,  2.15748723e-05,
+        1.08452242e-05, -1.36746652e-05, -5.56777430e-05, -1.17807768e-04,
+       -2.01126526e-04, -3.04675651e-04, -4.25179658e-04, -5.56926298e-04])
+    """
+
+    # Check for invalid input
+    if not isinstance(freq, int) and not isinstance(freq, float):
```

Comment:
```suggestion
    freq = float(freq)
```
This will error out nicely enough if something inappropriate is given